rest/route/service.go

```diff
@@ -130,19 +131,19 @@ func AttachHandler(app *gimlet.APIApp, opts HandlerOpts) {
 	app.AddRoute("/distros/{distro_id}").Version(2).Delete().Wrap(removeDistroSettings).RouteHandler(makeDeleteDistroByID())
 	app.AddRoute("/distros/{distro_id}").Version(2).Put().Wrap(createDistro).RouteHandler(makePutDistro())
 	app.AddRoute("/distros/{distro_id}/ami").Version(2).Get().Wrap(requireTask).RouteHandler(makeGetDistroAMI())
-	app.AddRoute("/distros/{distro_id}/client_urls").Version(2).Get().RouteHandler(makeGetDistroClientURLs(env))
+	app.AddRoute("/distros/{distro_id}/client_urls").Version(2).Get().Wrap(requireUserToggleable).RouteHandler(makeGetDistroClientURLs(env))
 	app.AddRoute("/distros/{distro_id}/execute").Version(2).Patch().Wrap(editHosts).RouteHandler(makeDistroExecute(env))
 	app.AddRoute("/distros/{distro_id}/icecream_config").Version(2).Patch().Wrap(editHosts).RouteHandler(makeDistroIcecreamConfig(env))
 	app.AddRoute("/distros/{distro_id}/setup").Version(2).Get().Wrap(editDistroSettings).RouteHandler(makeGetDistroSetup())
 	app.AddRoute("/distros/{distro_id}/setup").Version(2).Patch().Wrap(editDistroSettings).RouteHandler(makeChangeDistroSetup())
 
-	app.AddRoute("/hooks/github").Version(2).Post().RouteHandler(makeGithubHooksRoute(sc, opts.APIQueue, opts.GithubSecret, settings))
-	app.AddRoute("/hooks/aws").Version(2).Post().RouteHandler(makeEC2SNS(env, opts.APIQueue))
-	app.AddRoute("/hooks/aws/ecs").Version(2).Post().RouteHandler(makeECSSNS(env, opts.APIQueue))
+	app.AddRoute("/hooks/github").Version(2).Post().Wrap(requireUserToggleable).RouteHandler(makeGithubHooksRoute(sc, opts.APIQueue, opts.GithubSecret, settings))
```

Comment:
Hmm I'm actually concerned this will break our integrations 😬 Right now we validate in Parse for example https://github.com/evergreen-ci/evergreen/blob/e37538a7fd96246c2241148634adf2b4beabc438/rest/route/github.go#L80 but maybe that should be its own middleware so it doesn't look like these routes are unauthed? (Could be a split ticket though) Regardless though the average user isn't able to use these.

Reply:
Hmm ok - I'll remove the middleware for now.  I actually don't have a strong preference on whether we should validate via middleware or in `Parse`, since they are ultimately very similar concepts.  I'll add a ticket we can discuss in triage.

Reply:
https://jira.mongodb.org/browse/EVG-18002